ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/categories/AppCategoriesPresenterImpl.java

```diff
@@ -93,192 +90,140 @@ public int compare(AppCategory group1, AppCategory group2) {
+            public void onSuccess(List<AppCategory> result) {
+                filterCategories(result);
+                createWorkspaceTab(selectedAppCategory);
+                createHPCTab();
+                workspaceView.getTree().unmask();
+                hpcView.getTree().unmask();
             }
         });
-    }
-
-    @Override
-    public void onAppFavorited(AppFavoritedEvent appFavoritedEvent) {
-        final App app = appFavoritedEvent.getApp();
-        final AppCategory currentCategory = getSelectedAppCategory();
 
-        if (currentCategory == null || !FAVORITES.equals(currentCategory.getName())) {
-            // Adjust favorite category count.
-            final AppCategory favoriteCategory = findAppCategoryByName(FAVORITES);
-            int favCountAdjustment = app.isFavorite() ? 1 : -1;
-            updateAppCategoryAppCount(favoriteCategory, favoriteCategory.getAppCount() + favCountAdjustment);
-        }
     }
 
     @Override
-    public void onAppInfoSelected(final AppInfoSelectedEvent event) {
-        appDetailsDlgAsyncProvider.get(new AsyncCallback<AppDetailsDialog>() {
-            @Override
-            public void onFailure(Throwable caught) {
-                ErrorHandler.post(caught);
-            }
+    public void setViewDebugId(String baseID) {
+        workspaceView.asWidget().ensureDebugId(baseID + AppsModule.Ids.CATEGORIES);
+        hpcView.asWidget().ensureDebugId(baseID + AppsModule.Ids.HPC);
+    }
 
-            @Override
-            public void onSuccess(final AppDetailsDialog dlg) {
-
-                // Fetch details, otherwise App.getGroups may be null
-                appUserService.getAppDetails(event.getApp(), new AsyncCallback<App>() {
-                    @Override
-                    public void onFailure(final Throwable caught) {
-                        announcer.schedule(new ErrorAnnouncementConfig(appearance.fetchAppDetailsError(caught)));
-                    }
-
-                    @Override
-                    public void onSuccess(final App result) {
-                        // Create list of group hierarchies
-                        List<List<String>> appGroupHierarchies = Lists.newArrayList();
-                        for (AppCategory appCategory : result.getGroups()) {
-                            appGroupHierarchies.add(getGroupHierarchy(appCategory));
-                        }
-
-                        dlg.show(result,
-                                 searchRegexPattern,
-                                 appGroupHierarchies,
-                                 AppCategoriesPresenterImpl.this,
-                                 AppCategoriesPresenterImpl.this,
-                                 AppCategoriesPresenterImpl.this);
-                    }
-                });
-            }
-        });
+    void createHPCTab() {
+        hpcView.addAppCategorySelectedEventHandler(this);
+        addAppCategories(hpcView.getTree().getStore(), null, hpcCategories);
```

Comment:
If I need to get the entire treeStore create a better method for in the view instead of manipulating the tree directly.